fix: format and wording adjustments in workspaces, contributors and repo insights sections (#262)
open-sauced/docs: docs/maintainers/maintainers-guide.md

The Repository Insights feature allows you to:

- track the contributions of a specific team, a specific repository, or a group of repositories,
- view the number of PRs that are open, closed, in draft, and merged,
- view the average number of days it takes for a PR to be merged (PR velocity),
- identify the top contributors to your project,
- identify the current activity level and trends in the activity of your contributors.

### Creating a New Repository Insight Page

> To connect and track your repositories, click the "+" next to "Insights" in the sidebar, then select "New Repository Insight". You will be redirected to a page where you can create your new Repository Insight Page.

Reviewer: ```suggestion
To connect and track your repositories, click the "+" next to "Insights" in the sidebar, and select "New Repository Insight". From there, you will be redirected to a page where you can create your new Repository Insight Page.
```

### Using Your Repository Insights Page

> Once you have connected your repositories, you will be redirected to your Repository Insight Page. Here, you will see a dashboard with an overview of the repositories and the contributors who have contributed to them.

Author: Can you help to double check this with me?

Because after I created a repository insights page, it redirected me to the "Personal Workspace" instead of the new created page as screen recording below.

![create new repo insights 1](https://github.com/open-sauced/docs/assets/45172775/015f47d4-d9bd-4ad1-9059-1fbb81081f9b)

Reviewer 2: Just created a bug report for that.

There are three tabs that provide you with more information on these repositories:

> #### Reports

Author: I don't see "Reports" tab in on the Repository Insight Page. There are only "Dashboard", "Contributors" and "Activity".

![Screenshot 2024-02-28 215744](https://github.com/open-sauced/docs/assets/45172775/634ff737-6a81-4cdd-9f7e-84d33e06489c)

Reviewer: Hi @adiati98. Repositories section does sound clearer.

Author: Repositories Dashboard sounds great!

Also, I'll remove Reports altogether with this PR. 👍

The "Reports" tab allows paid users to filter for more information about their repositories over the last 30 days and to generate a Downloadable CSV. Filters include: Top 100 Repos, Minimum 5 Contributors, Recent, Most Active, and Spammed.

#### Repositories

The "Repositories" tab allows you to view more detailed information on each repository, including Activity, PR Overview, PR Velocity, Spam, Contributors, and activity over the last 30 days. To learn more about these features, see [Understanding Repository Insights Data](understanding-repo-insights.md).

![repositories tab](../../static/img/repos-insights.png)

#### Contributors

The "Contributors" tab allows you to view more detailed information on each contributor, including Activity, Repositories, the date of their last contribution, Time Zone, and the number of contributions.

> You can select and add your contributors to a Contributor Insight Page to learn more about them.

Author: Should we link the contributor insight page to the next section or leave it as is?

```suggestion
You can select and add your contributors to a [Contributor Insight Page](#contributor-insights-connecting-with-contributors) to learn more about them.
```

To learn and understand more about the data provided, see [Understanding Repository Insights Data](understanding-repo-insights.md).

![repositories tab](../../static/img/repos-insights.png)

#### Contributors

The "Contributors" tab allows you to view more detailed information on each contributor, including activity levels, the number of repositories they contributed to, the date of their last contribution, their most used language, their time zone, the number of contributions, and their activity stats over the last 30 days.

You can select and add your contributors to a [Contributor Insight Page](#contributor-insights-connecting-with-contributors) to learn more about them.

There are two ways to add selected contributors to a Contributor Insight Page:

1. Create a new page.
2. Add them to an existing page.

![contributors insight list](../../static/img/contributors-insights.png)

#### Activity

The "Activity" tab shows a graph of the last time each contributor created their PR and how many lines of code they've touched.

> It also provides more detailed information on each contributor when you hover over their image, including their latest PRs and repositories they've contributed to. Clicking their image will bring you to their profile on OpenSauced. You can use the information in this tab to help you learn about their engagement and general interests. It will be helpful if you want to collaborate with them or consider recruiting a maintainer for your project.

Author: This is the "Activity" tab subsection that I added for the Maintainers Guide. Same wording, different purpose that's shown in the last two sentences here.

Reviewer 2: Sounds good.

- **VIP contributors**: Highlight and track contributors you've identified as key maintainers or significant contributors.
- **Sales leads**: Identify contributors whose engagement could potentially be turned into sales opportunities.
- **Potential hires**: Observe contributors who exhibit skill and commitment, making them prime candidates for hiring.
- **Alumni contributors**: Keep an eye on contributors who have decreased their engagement or left the project.

### Creating a New Contributor Insight Page

1. Click the "+" next to "Insights" in the sidebar.
2. Click "New Contributor Insight." You will be redirected to a page where you can create your new Contributor Insight Page.

   ![New Contributor Insight Page](../../static/img/contrib-insight-new.png)

3. Give your page a name.
4. Choose your page visibility. You can choose to make your Contributor Insight Page public or private. By default, it is set to public.
5. Add contributors to your page by searching for their GitHub username, syncing your GitHub Team, or importing your GitHub Following.

   ![team sync gif](../../static/gif/team-sync.gif)

> ## Using Your Contributor Insight Page

Author: In this section, there are three tabs explained: Overview, Activity, and Contributor.
With the new workspace, we now have: Overview, Activity, and Highlights (see screenshot below).

We might want to update this as well as add a new image to replace the one in "Activity" subsection. We can add it here or create a new issue for that.

![Screenshot 2024-03-03 234339](https://github.com/open-sauced/docs/assets/45172775/c45c665f-0f3e-4cfd-b8ea-9ac042ffa774)

Reviewer: > In this section, there are three tabs explained: Overview, Activity, and Contributor.
> 
> With the new workspace, we now have: Overview, Activity, and Highlights (see screenshot below).
> 
> 
> 
> We might want to update this as well as add a new image to replace the one in "Activity" subsection. We can add it here or create a new issue for that.
> 
> 
> 
> ![Screenshot 2024-03-03 234339](https://github.com/open-sauced/docs/assets/45172775/c45c665f-0f3e-4cfd-b8ea-9ac042ffa774)
> 
> 
> 
> 
> 
> 

I think creating a separate issue for this would be good @adiati98. If you & @BekahHW go with this idea, I'll create the issue for it.

Reviewer 2: I would rather have a new issue for this, just so we can get this merged in.

Reviewer: Ok I'll create the issue

Author: @CBID2 I've changed this to "Highlights".
I just noticed we have it in the contributor section. So, I use that one.

Reviewer: Hey @adiati98. I just saw this message. I made [the issue](https://github.com/open-sauced/docs/issues/270). Since we don't need it yet, I kept it in triage mode. Is that ok?

- **Interacting contributors**: Keep tabs on contributors actively interacting with your repository.
- **Organization's developers**: Monitor the contributions and performance of your internal team.
- **VIP contributors**: Highlight and track contributors you've identified as key maintainers or significant contributors.
- **Sales leads**: Identify contributors whose engagement could potentially be turned into sales opportunities.
- **Potential hires**: Observe contributors who exhibit skill and commitment, making them prime candidates for hiring.
- **Alumni contributors**: Keep an eye on contributors who have decreased their engagement or left the project.

### Creating a New Contributor Insight Page

1. Click the "+" next to "Insights" in the sidebar.
2. Click "New Contributor Insight." You will be redirected to a page where you can create your new Contributor Insight Page.

   ![New Contributor Insight Page](../../static/img/contrib-insight-new.png)

3. Give your page a name.
4. Add contributors to your page by searching for their GitHub username, syncing your GitHub Team, or importing your GitHub Following.

> 5. Choose your page visibility. You can choose to make your Contributor Insight Page public or private. By default, it is set to public.

Author: ```suggestion
5. Once your page is created, you can edit and choose to make it public or private. By default, it is set to public. To create a private page, you can upgrade your account to a paid plan.
```

1. Create a new Contributor Insight Page with selected contributors.

> 2. Add selected contributors to an existing Contributor Insight Page.

Author: I'm thinking to move line 68-71 to before the image so it won't be missed. Thoughts?

Reviewer: That would help @adiati98. It'll make the instructions clearer for contributors.

## Contributor Insights: Connecting with Contributors

OpenSauced Contributor Insights feature allows you to categorize, monitor, and analyze various groups of contributors. You can use Contributor Insights to track the contributions of a specific team, a specific repository, or a group of repositories. This feature offers granular insights into each contributor's activities and contributions, helping maintainers to track individual contribution histories, identify active engagement, recognize new or alumni contributors, and compare the performance of different contributors.

The Contributor Insights features allow you to:

- track the commits and PRs of the contributors on your list,
- filter your contributors by their activity level, including most active, new, and alumni,

> - view the repositories your contributors have contributed to, the top programming languages they've used, and their time zones.

Reviewer: [improvement] the wording here could be clearer. Thoughts?
```suggestion
- view the repositories your contributors have worked on, the top programming languages they've used, and their time zones.
```

Author: I think "contributed to" is make much more sense here.

Reviewer: I don't know @adiati98. Saying "contributors have contributed to" does sound a bit redundant, so using a different word like "worked on" conveys the same point but in a clearer way. What do you think @BekahHW?

### Why Create a Contributor Insight Page?

There are many reasons to create a Contributor Insight Page, but here's some inspiration to get you started:

- **Interacting contributors**: Keep tabs on contributors actively interacting with your repository.
- **Organization's developers**: Monitor the contributions and performance of your internal team.
- **VIP contributors**: Highlight and track contributors you've identified as key maintainers or significant contributors.
- **Sales leads**: Identify contributors whose engagement could potentially be turned into sales opportunities.
- **Potential hires**: Observe contributors who exhibit skill and commitment, making them prime candidates for hiring.
- **Alumni contributors**: Keep an eye on contributors who have decreased their engagement or left the project.

### Creating a New Contributor Insight Page

1. Click the "+" next to "Insights" in the sidebar.

> 2. Click "New Contributor Insight". You will be redirected to a page where you can create your new Contributor Insight Page.

Reviewer: ```suggestion
2. Click "New Contributor Insight". From there, you will be redirected to a page where you can create your new Contributor Insight Page.
```

Reviewer 2: @CBID2 punctuation should go inside the quotes. I also don't feel this updates are necessary.

   ![New Contributor Insight Page](../../static/img/contrib-insight-new.png)

3. Give your page a name.

> 4. Choose your page visibility. You can choose to make your Contributor Insight Page public or private. By default, it is set to public.

Reviewer: ```suggestion
4. Choose your page visibility. By default, your Contributors Insight page is set to public, but can you make it private.
```

Author: I feel like the original is more clear and in the flow here.
But I'll let @BekahHW decide. 🙂

Also, it supposed to be "Contributor Insight Page", and not "Contributors Insight page".

Reviewer 2: These are fine as they stand.

### Overview

The "Overview" tab gives you a high-level view of the contributors in your Contributor Insight Page, including the total number of commits in the last 30 days and the types of contributors: Active, New, and Alumni.

### Activity

> The "Activity" tab gives you a graph view with more detailed information on each contributor, including the type of activity, repositories they've contributed to, and how they compare to each other.

Reviewer: ```suggestion
The "Activity" tab gives you a graph with detailed information on each contributor, including the type of activity, repositories they've contributed to, and how they compare to each other.
```

> You can filter your Contributor Insight Page by All Contributors, Active Contributors, New Contributors, and Alumni Contributors.

Reviewer: ```suggestion
You can filter your Contributor Insight Page by choosing All Contributors, Active Contributors, New Contributors, or Alumni Contributors.
```

There are two ways to create a new highlight:

> 1. After clicking the "[Highlights](https://app.opensauced.pizza/feed)" tab in the top navigation, click the input that says "Post a highlight to show your work!"

Author: If you see the app, the input does say "Post a highlight to show your work!"

![Screenshot 2024-02-29 200329](https://github.com/open-sauced/docs/assets/45172775/4735114e-a3c6-4f3f-a9fb-872f548f540a)

> 2. From your profile, click the input that says "Post a highlight to show your work!"

Reviewer: [suggestion] We mentioned this part in the previous step, so I think it is best to omit this part. Thoughts?
```suggestion
```

![highlight suggestions](../../static/img/highlights.png)

:::note

> If you're adding a blog post, you'll need to associate it to a repository by clicking the "Add a repo" button.

Reviewer 2: @CBID2 there's an issue and discussion in the app repo about connecting it that you can check out.
